airbyte-integrations/connectors/source-google-analytics-v4/source_google_analytics_v4/source.py

```diff
@@ -0,0 +1,516 @@
+                                ]
+                            }
+                        ],
+                        "minimums": [
+                            {
+                                "values": [
+                                    "98"
+                                ]
+                            }
+                        ],
+                        "rowCount": 1,
+                        "rows": [
+                            {
+                                "metrics": [
+                                    {
+                                        "values": [
+                                            "98"
+                                        ]
+                                    }
+                                ]
+                            }
+                        ],
+                        "totals": [
+                            {
+                                "values": [
+                                    "98"
+                                ]
+                            }
+                        ]
+                    }
+                }
+            ]
+        }
+
+        Return record which is a map of metric and dimension names and values, like:
+
+        record = {
+                    "view_id":"1111111"
+                    "ga_date":"20210212",
+                    "ga_users":3,
+                    "ga_newUsers":2,
+                    "ga_sessions":7,
+                    "ga_sessionsPerUser":8.0,
+                    "ga_avgSessionDuration":201.0,
+                    "ga_pageviews":43,
+                    "ga_pageviewsPerSession":12.5,
+                    "ga_avgTimeOnPage":83.14035087719298,
+                    "ga_bounceRate":0.0,
+                    "ga_exitRate":6.523809523809524
+        }
+        """
+        json_response = response.json()
+        reports = json_response.get(self.report_field, [])
+
+        for report in reports:
+            column_header = report.get("columnHeader", {})
+            dimension_headers = column_header.get("dimensions", [])
+            metric_headers = column_header.get("metricHeader", {}).get("metricHeaderEntries", [])
+
+            for row in self.get_data(report):
+                record = {}
+                dimensions = row.get("dimensions", [])
+                metrics = row.get("metrics", [])
+
+                for header, dimension in zip(dimension_headers, dimensions):
+                    data_type = self.lookup_data_type("dimension", header)
+
+                    if data_type == "integer":
+                        value = int(dimension)
+                    elif data_type == "number":
+                        value = float(dimension)
+                    else:
+                        value = dimension
+
+                    record[header.replace("ga:", "ga_")] = value
+
+                for i, values in enumerate(metrics):
+                    for metric_header, value in zip(metric_headers, values.get("values")):
```

Comment:
same Q here on order of metric_headers and values.get("values")

Reply:
Yes.